### ISSUE TITLE
Preview changes not being picked up

### ISSUE THREAD

**Issue by ccorda** [patronage/bubs-next#192]
The first time I preview a draft post, it reflects what is in WP. But subsequent changes and re-preview isn't showing an updated page.

Here is a sample page:
https://bubsnext.wpengine.com/wp-admin/post.php?post=3&action=edit

When using the graphql IDE with a sample query, it does correctly update each time:

query:
```
query GetContent($slug: ID!, $preview: Boolean) {
  contentNode(id: $slug, idType: URI, asPreview: $preview) {
    __typename
    id
    databaseId
    isPreview
    link
    slug
    uri
    date
    previewRevisionDatabaseId
    previewRevisionId
    ... on Page {
      id
      content
    }
  }
}

```

variables:
```
{
  "slug": "/?page_id=81",
  "preview": true
}
```

I've tested this on a couple sites, both have graph cdn on, so it's possible the caching is happening there, though when I purge cache it doesn't seem to correct anything.

**Comment by ccorda** [comment]
This was a graphcdn issue, the `AUTHENTICATED` scope was causing the page to be cached.

<img width="551" alt="Screen Shot 2021-12-30 at 8 10 28 PM" src="https://user-images.githubusercontent.com/45271/147797399-1af4e61a-6916-411a-8d32-24660c2d8696.png">

Need to make sure we document this when we commit the docs for #143.

**Comment by ccorda** [comment]
Via support request from Graph CDN, apparently there isn't an easy way on their end to do this currently. They are tracking the request here:
https://graphcdn.canny.io/feature-requests/p/header-to-disable-caching

They suggested one alternative, and i can think of a few other:

1) Store graphcdn endpoint and origin separately as two ENV vars, and conditionally always load from origin on authenticated requests

2) Have our webhooks code expire cache whenever preview is updated

3) Have our origin send a no-cache header for authorized requests

**Comment by ccorda** [comment]
Turns out graphcdn was a red herring. WP Graphql is supposed to send [no-cache headers](https://github.com/wp-graphql/wp-graphql/issues), and it does. Graph CDN is then supposed to not save anything to cache, and it doesn't.

The actual issue is something in our WP logic. When the [sample post](https://bubsnext.wpengine.com/wp-admin/post.php?post=3&action=edit) is published, then each subsequent save generates a new preview url: `/92`, `/93`, etc.

However when it's a draft, it always goes to `/3`, which loads the last saved version of the post, but not the temporary preview ones.

I think the solve would be on the `headless-redirect.php` side, to send the user to the preview ID regardless of post publish status.

**Comment by ccorda** [comment]
After further testing, it was hard (and I don't believe possible) to setup roles to keep caches for public and authenticated requests on graphcdn.

While the fix in #193 of hitting the origin direct did work, this might be better:
https://graphcdn.canny.io/changelog/define-headers-to-bypass-the-graphcdn-edge-cache

**Comment by ccorda** [comment]
Solved by bypassing graphcdn if logged in to preview mode:
https://github.com/patronage/bubs-next/blob/942cac6ebc97ce50fe60b9edeb0566f26f8e23ca/docs/performance-backend.md#enabling-graphcdn
